mongoengine/fields.py

```diff
@@ -941,6 +941,19 @@ def __get__(self, instance, owner):
             and value
         ):
             instance._data[self.name] = [self.field.build_lazyref(x) for x in value]
+        elif (
```

Comment:
It sounds weird to have to add dependency to EnumField in the ListField class, please add a test case that highlights the bug so I can see if this is the right place to add some code

Reply:
I added a test case but you're right that it might not be the way to fix it.
In the test case if I instanciate `Model(statuses=['new'])` instead of `Model(statuses=[Status.NEW])` the value doesn't get casted and a `ValidationError` is raised (which isn't the case if I instanciate `Model(status='new')`) 

So I'm thinking somewhere in the `EnumField` code or, most probably, in the way `ListField` process its data, something is missing or just a bit off.

Reply:
Fixed the test case, got the cast better handled. Linting is still a pain x)